### PR TITLE
KOGITO-5816 - Enable Native test execution for process and add missin…

### PR DESCRIPTION
**Pull request by cristianonicolai**
…g classes to native reflection

### CODE REVIEW

**Comment by cristianonicolai** [comment]
Jenkins run native

File: quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common-deployment/src/main/java/org/kie/kogito/quarkus/common/deployment/KogitoAssetsProcessor.java [hunk 1]
```diff
@@ -134,6 +134,14 @@ public ReflectiveClassBuildItem reflectionJobsManagement() {
         return new ReflectiveClassBuildItem(true, true, "org.kie.kogito.jobs.api.Job");
     }
 
+    @BuildStep
```
Comment (tiagodolphine) [inline quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common-deployment/src/main/java/org/kie/kogito/quarkus/common/deployment/KogitoAssetsProcessor.java:137]: if this is just needed for processes, I think it can be moved to ProcessAssetsProcessor, like this one https://github.com/kiegroup/kogito-runtimes/blob/7e1cb1ea4d3e4802a039233f3db5f90851280c4a/quarkus/extensions/kogito-quarkus-processes-extension/kogito-quarkus-processes-deployment/src/main/java/org/kie/kogito/quarkus/processes/deployment/ProcessesAssetsProcessor.java#L136-L139

**Review by tiagodolphine — APPROVED** [review]

File: quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common-deployment/src/main/java/org/kie/kogito/quarkus/common/deployment/KogitoAssetsProcessor.java (hunk 1 above)
Comment (cristianonicolai) [reply quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common-deployment/src/main/java/org/kie/kogito/quarkus/common/deployment/KogitoAssetsProcessor.java:137]: makes sense to me, but should we also move _reflectionJobsManagement_ ?

**Comment by cristianonicolai** [comment]
Jenkins run native

**Review by ricardozanini — APPROVED** [review]

**Comment by cristianonicolai** [comment]
Jenkins run native

File: integration-tests/integration-tests-quarkus-norest/src/test/java/org/kie/kogito/quarkus/NativeNoEndpointIT.java [hunk 2]
```diff
@@ -1,24 +0,0 @@
...
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.kie.kogito.quarkus;
-
-import io.quarkus.test.junit.NativeImageTest;
-
-@NativeImageTest
-public class NativeNoEndpointIT extends NoEndpointTest {
```
Comment (danielezonca) [inline integration-tests/integration-tests-quarkus-norest/src/test/java/org/kie/kogito/quarkus/NativeNoEndpointIT.java:21]: Why this change? Is this native test broken?

Comment (cristianonicolai) [reply integration-tests/integration-tests-quarkus-norest/src/test/java/org/kie/kogito/quarkus/NativeNoEndpointIT.java:21]: @danielezonca replace with `@QuarksuIntegrationTest` as it can reuse the same class for any build type, native or jvm. That's something we should look revisiting in the future as it simplifies the Maven configuration.

Comment (danielezonca) [reply integration-tests/integration-tests-quarkus-norest/src/test/java/org/kie/kogito/quarkus/NativeNoEndpointIT.java:21]: Ok good to know so this mean we might revisit and remove all `@NativeImageTest` right?

Comment (cristianonicolai) [reply integration-tests/integration-tests-quarkus-norest/src/test/java/org/kie/kogito/quarkus/NativeNoEndpointIT.java:21]: Correct, its deprecated in Quarkus as well

**Comment by radtriste** [comment]
jenkins rerun examples native

File: integration-tests/integration-tests-quarkus-processes/pom.xml [hunk 3]
```diff
@@ -160,27 +160,23 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <systemPropertyVariables combine.children="append">
+            <container.image.kafka>${container.image.kafka}</container.image.kafka>
```
Comment (radtriste) [inline integration-tests/integration-tests-quarkus-processes/pom.xml:168]: do we really need a container image for unit testing ?

Comment (cristianonicolai) [reply integration-tests/integration-tests-quarkus-processes/pom.xml:168]: its used by unit test set as `@QuarkusTest` in the project

Comment (radtriste) [reply integration-tests/integration-tests-quarkus-processes/pom.xml:168]: My point is that unit test should not have external dependency.
If it needs, then it should be moved at least to integration tests

Comment (cristianonicolai) [reply integration-tests/integration-tests-quarkus-processes/pom.xml]: np, end up refactoring the test

**Comment by cristianonicolai** [comment]
Jenkins run native

**Review by radtriste — APPROVED** [review]

**Comment by cristianonicolai** [comment]
Jenkins rerun examples tests

**Comment by cristianonicolai** [comment]
Jenkins rerun examples tests
